src/controller/python/chip/yaml/parser.py

```diff
@@ -170,18 +337,28 @@ def __init__(self, item: dict, cluster: str, data_model_lookup: DataModelLookup)
             raise UnexpectedParsingError(
                 f'ReadAttribute doesnt have valid attribute_type. {self.label}')
 
+        if 'optional' in item:
+            self._possibly_unsupported = True
+
         self._expected_raw_response = item.get('response')
         if (self._expected_raw_response is None):
             raise UnexpectedParsingError(f'ReadAttribute missing expected response. {self.label}')
 
-        if 'optional' in item:
-            self._possibly_unsupported = True
+        self._save_as = _SaveAs.extract_save_as_property(
```

Comment:
This saveas implementation looks like it's for the response as a whole. I realize this is not implemented yet, but will this system also work for saving struct members?

Reply:
Actually, looking more, it's not stucts I was thinking of, but command responses. Similar argument, except that the yaml lacks a way to express this for structs. The reasoning is the same though - because the yaml doesn't have a good way to specify the fields of a structure, you wouldn't be able to use the individual fields of a struct or command response. We don't have parsing for a dot notation, so for command responses you can set the saveAs: label for each field. 

You can see an example in the TestSaveAs tests "Send Test Add Arguments Command" (first test), where the saveAs: label is applied to one of list members in a values: list as opposed to directly in the response. 

    - label: "Send Test Add Arguments Command"
      command: "testAddArguments"
      arguments:
          values:
              - name: "arg1"
                value: 3
              - name: "arg2"
                value: 17
      response:
          values:
              - name: "returnValue"
                saveAs: TestAddArgumentDefaultValue
                value: 20

I'm actually not sure what the right answer is for structs - we either need to allow the individual fields to be saved, or we need way to specify the field from a saved variable. We don't have tests defined for either case currently.

Reply:
Yeah we won't have issues saving a struct from command response going foward.

I actually need to do a little bit of a refactor to get this test working specifically. That refactor will come as a standalone PR to make reviewing more simple